src/tests/PubSub.test.ts

```diff
@@ -0,0 +1,252 @@
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(isNewSpy).toHaveBeenCalledTimes(1);
+      expect(isNewSpy).toHaveBeenCalledWith({ test: "test" }, "test");
+      expect((pubSub as any).client.publishJSON).not.toHaveBeenCalled();
+    });
+
+    it("should call hashData", async () => {
+      isNewSpy.mockRestore();
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(hashDataSpy).toHaveBeenCalledTimes(1);
+      expect(hashDataSpy).toHaveBeenCalledWith({ test: "test" });
+      isNewSpy = jest.spyOn(PubSub.prototype as any, "isNew");
+    });
+
+    it("should save MessagesSent", async () => {
+      isNewSpy.mockResolvedValueOnce(true);
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect(typeorm.getManager).toHaveBeenCalledTimes(1);
+      expect(typeorm.getManager().getRepository).toHaveBeenCalledTimes(1);
+      expect(typeorm.getManager().getRepository).toHaveBeenCalledWith(
+        MessagesSent
+      );
+      expect(
+        typeorm.getManager().getRepository(MessagesSent).save
+      ).toHaveBeenCalledTimes(1);
+    });
+
+    it("should do send the message to Pub/Sub", async () => {
+      isNewSpy.mockResolvedValueOnce(true);
+      const pubSub = new PubSub("test");
+      await pubSub.publish(
+        "test",
+        { test: "test" },
+        {
+          schema:
+            "https://api.thegraph.com/subgraphs/name/aragon/aragon-govern-rinkeby",
+          interval: 60000,
+          additionalFields: {
+            network: "rinkeby",
+          },
+        }
+      );
+      expect((pubSub as any).client.publishJSON).toHaveBeenCalledTimes(1);
+      expect((pubSub as any).client.publishJSON).toHaveBeenLastCalledWith({
+        source: "graphql-events",
+        type: "test",
+        additionalFields: {
+          network: "rinkeby",
+        },
+        message: { test: "test" },
+      });
+    });
+  });
+
+  describe("isNew", () => {
+    it("should return false if no message is given", async () => {
+      // set to any because isNew is a private function
+      const pubSub = new PubSub("test") as any;
+      expect(await pubSub.isNew(undefined, "test")).toBe(false);
+    });
+
+    it("should call hashData", async () => {
+      // set to any because isNew is a private function
+      const pubSub = new PubSub("test") as any;
+      pubSub.hashData = jest.fn();
```

Comment:
here I don't care about the real implementation of this function because I only mock the function from the object created the line before and not from the prototype